tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts

```diff
@@ -465,3 +466,19 @@ export function getSafeFilenameForName(apiItemName: string): string {
 	// https://github.com/microsoft/rushstack/issues/1308
 	return apiItemName.replace(badFilenameCharsRegExp, "_").toLowerCase();
 }
+
+/**
+ * Extracts the text from the provided excerpt and adjusts it to be on a single line, and to omit any trailing `;`.
+ * @remarks Useful when a shortened version of a code excerpt is wanted, and especially if the contents need to fit on a single line.
+ * @public
+ */
+export function getSingleLineExcerptText(excerpt: Excerpt): string {
+	// Regex replaces line breaks with spaces to ensure everything ends up on a single line.
+	let signatureExcerpt = excerpt.text.trim().replace(/\s+/g, " ");
+
+	if (signatureExcerpt.endsWith(";")) {
+		signatureExcerpt = signatureExcerpt.slice(0, signatureExcerpt.length - 1);
+	}
```

Comment:
So this function should be used just for code excerpts that are single-statement? If we had something like

```typescript
const a = 1;
const b = 2;
```

The transformed version would be `const a = 1; const b = 2` which feels a bit weird with the semicolon in the middle but none at the end.

Which I don't mind, but maybe worth calling out in remarks.

Reply:
That's an interesting point. That policy probably doesn't belong in this helper, if it's going to be public. Consumers can still trim off the trailing semi if they want. Though I want to double check what scenarios that's even relevant for.

Reply:
I updated the base PR to omit the `;` trimming from the public function. The 2 internal use-cases still want it (specifically for heading text and table entry text for various signature types. Not critical in those scenarios, but nice to have.